packages/worker/src/worker-options.ts

```diff
@@ -1090,14 +1092,20 @@ export function toNativeWorkerOptions(opts: CompiledWorkerOptionsWithBuildId): n
     workflowTaskPollerBehavior: toNativeTaskPollerBehavior(opts.workflowTaskPollerBehavior),
     activityTaskPollerBehavior: toNativeTaskPollerBehavior(opts.activityTaskPollerBehavior),
     nexusTaskPollerBehavior: toNativeTaskPollerBehavior(opts.nexusTaskPollerBehavior),
-    enableNonLocalActivities: opts.enableNonLocalActivities,
+    taskTypes: {
+      enableWorkflows,
+      enableLocalActivities,
+      enableRemoteActivities: opts.enableNonLocalActivities && opts.activities.size > 0,
```

Comment:
That's basically saying:

```
enableRemoteActivities: (enableWorkflows && opts.activities.size > 0) && opts.activities.size > 0
```

Which is incorrect. We don't need workflows to enable remote activities. I'm surprised tests passed with this.